src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Tests/VTableGCHandlePair.cs

```diff
@@ -2,15 +2,16 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
 
 namespace ComInterfaceGenerator.Tests
 {
-    public static unsafe class VTableGCHandlePair<TUnmanagedInterface>
-        where TUnmanagedInterface : IUnmanagedInterfaceType<TUnmanagedInterface>
+    public unsafe class VTableGCHandlePair<TUnmanagedInterface> : IUnmanagedObjectUnwrapper
+        where TUnmanagedInterface : IUnmanagedInterfaceType
     {
         public static void* Allocate(TUnmanagedInterface obj)
         {
-            void** unmanaged = (void**)NativeMemory.Alloc((nuint)sizeof(void*) * (nuint)IUnmanagedVirtualMethodTableProvider.GetVirtualMethodTableLength<TUnmanagedInterface>());
+            void** unmanaged = (void**)NativeMemory.Alloc((nuint)sizeof(void*) * 2);
```

Comment:
Instead of hard-coding this, pass it in as a parameter? It'll make this type a little more useful if we use it in more testing scenarios.

Reply:
Doesn't this allocation allocate for the pair, and not the VTable length? Or am I misunderstanding?

Reply:
You are correct! I messed this up previously.